allennlp/modules/conditional_random_field.py

```diff
@@ -1,13 +1,57 @@
 """
 Conditional random field
 """
-from typing import List
+from typing import List, Tuple, Dict
 
 import torch
 from torch.autograd import Variable
 
+from allennlp.common.checks import ConfigurationError
 import allennlp.nn.util as util
 
+
+def allowed_transitions(constraint_type: str, tokens: Dict[int, str]) -> List[Tuple[int, int]]:
```

Comment:
A short doctring describing the two constraints and `tokens` would be good.

Reply:
done